src/nvim/ex_getln.c

```diff
@@ -4308,11 +4306,12 @@ in_history (
       last_i = i;
     }
     history[type][i].hisnum = ++hisnum[type];
-    history[type][i].viminfo = FALSE;
     history[type][i].hisstr = str;
-    return TRUE;
+    history[type][i].timestamp = os_time();
+    history[type][i].additional_elements = NULL;
```

Comment:
And also because garbagecollect(1) calls with `-DEXITFREE` are always performed.